core/src/com/unciv/ui/pickerscreens/TechPickerScreen.kt

```diff
@@ -95,30 +96,30 @@ class TechPickerScreen(internal val civInfo: CivilizationInfo, centerOnTech: Tec
         val allTechs = civInfo.gameInfo.ruleSet.technologies.values
         if (allTechs.isEmpty()) return
         val columns = allTechs.map { it.column!!.columnNumber }.maxOrNull()!! + 1
-        val rows = allTechs.map { it.row }.maxOrNull()!! + 1
+        val rows = allTechs.map { it.row }.maxOrNull()!!
         val techMatrix = Array<Array<Technology?>>(columns) { arrayOfNulls(rows) } // Divided into columns, then rows
+        val erasNamesToColumns = LinkedHashMap<String, ArrayList<Int>>()
 
-        for (technology in allTechs) {
+        allTechs.forEach { technology ->
             techMatrix[technology.column!!.columnNumber][technology.row - 1] = technology
-        }
 
-        val erasNamesToColumns = LinkedHashMap<String, ArrayList<Int>>()
-        for (tech in allTechs) {
-            val era = tech.era()
+            val era = technology.era()
             if (!erasNamesToColumns.containsKey(era)) erasNamesToColumns[era] = ArrayList()
-            val columnNumber = tech.column!!.columnNumber
-            if (!erasNamesToColumns[era]!!.contains(columnNumber)) erasNamesToColumns[era]!!.add(columnNumber)
+            val columnNumber = technology.column!!.columnNumber
+            if (!erasNamesToColumns[era]!!.contains(columnNumber))
+                erasNamesToColumns[era]!!.add(columnNumber)
         }
         var i = 0
-        for ((era, columns) in erasNamesToColumns) {
+        erasNamesToColumns.forEach { (era, columns) ->
```

Comment:
No minimum API level warning? There was one in one of those collections...

Reply:
Even without the minimum API, for() is better than forEach() for debug and readabillity (although readability here is debatable)